src/material-experimental/mdc-form-field/_mdc-text-field-structure-overrides.scss

```
@@ -27,14 +27,14 @@
     border: none;
   }
 
-  // Reset the padding that MDC sets on native input elements. We cannot rely on this
+  // Reset the height that MDC sets on native input elements. We cannot rely on this
   // spacing as we support arbitrary form-field controls which aren't necessarily matching
   // the "mdc-text-field__input" class. Note: We need the first selector to overwrite the
   // default no-label MDC padding styles which are set with a very high specificity.
```

We should update the comment to explain the new state. e.g.

```suggestion
  // Reset the height that MDC sets on native input elements. We cannot rely on their
  // fixed height as we handle vertical spacing differently. MDC sets a fixed height for
  // inputs and modifies the baseline so that the textfield matches the spec. This does
  // not work for us since we support arbitrary form field controls which don't necessarily
  // use an input element. We organize the vertical spacing on the infix container.
```

Something like that? Might be also worth checking if the high specificity can be removed now that no padding is overriden?